pkg/health/condition/check_ready.go

```diff
@@ -31,16 +31,27 @@ func (r *readyCheck) Check(status druidv1alpha1.EtcdStatus) Result {
 		}
 	}
 
+	// TODO: remove this case as soon as leases are completely supported by etcd-backup-restore
+	if len(status.Members) == 0 {
+		return &result{
+			conType: druidv1alpha1.ConditionTypeReady,
+			status:  druidv1alpha1.ConditionUnknown,
+			reason:  "NoMembersInStatus",
+			message: "Cannot determine readiness since status has no members",
+		}
+	}
+
 	var (
 		size         = utils.Max(int(*status.ClusterSize), len(status.Members))
 		quorum       = size/2 + 1
 		readyMembers = 0
 	)
 
 	for _, member := range status.Members {
-		if member.Status == druidv1alpha1.EtcdMemeberStatusReady {
-			readyMembers++
+		if member.Status == druidv1alpha1.EtcdMemeberStatusNotReady {
+			continue
```

Comment:
This change would mean unknown members will still be counted un `readyMembers` and hence into readiness check. This is a deviation from the previous implementation but seems to be correct. 👍

Reply:
I noticed it while testing when UNKNOWN members caused the `Ready` condition to become `false` very early. Tolerating UNKNOWN might make sense as I see it today but we can still change it in the future if it turns out to be too tolerant.